geti_sdk/rest_clients/project_client/project_client.py

```diff
@@ -100,24 +97,60 @@ def get_all_projects(
             for project in project_rest_list
         ]
         if get_project_details:
-            return [self._get_project_detail(project) for project in project_list]
+            project_detail_list: List[Project] = []
+            for project in project_list:
+                try:
+                    project_detail_list.append(self._get_project_detail(project))
+                except GetiRequestException as e:
+                    if e.status_code == 403:
+                        logging.info(
+                            f"Unable to access project `{project.name}` details: Unauthorized."
+                        )
+                        project_detail_list.append(project)
+            return project_detail_list
         else:
             return project_list
 
-    def get_project_by_name(self, project_name: str) -> Optional[Project]:
+    def get_project_by_name(
+        self, project_name: str, project_id: Optional[str] = None
+    ) -> Optional[Project]:
         """
         Get a project from the Intel® Geti™ server by project_name.
 
         :param project_name: Name of the project to get
+        :param project_id: Optional ID of the project to get. Only used if more than
+            one project named `project_name` exists in the workspace.
+        :raises: ValueError in case multiple projects with the specified name exist on
+            the server, and no `project_id` is provided in order to allow unique
+            identification of the project.
         :return: Project object containing the data of the project, if the project is
             found on the server. Returns None if the project doesn't exist
         """
         project_list = self.get_all_projects(get_project_details=False)
-        project_entry = next(
-            (project for project in project_list if project.name == project_name), None
-        )
-        if project_entry is not None:
-            return self._get_project_detail(project_entry)
+        matches = [project for project in project_list if project.name == project_name]
+        if len(matches) == 1:
+            return self._get_project_detail(matches[0])
+        elif len(matches) > 1:
+            if project_id is None:
+                raise ValueError(
```

Comment:
it would be helpful to provide some additional information to the user. For sure the list of project ids alongside their creation dates, possibly types